lib/condition/condition.ts

```diff
@@ -167,58 +158,20 @@ export class Range {
                     this.min = value;
                 }
                 return;
-            case TREE.custom("InBetweenRelation"):
-                if (dataType === XSD.custom("gYear")) {
-                    const result = this.gYearToMinMax(value);
-                    if (!result) return;
-                    const [min, max] = result;
-                    if (!this.min || min <= this.min) {
-                        this.min = min;
-                        this.eqMin = true;
-                    }
-                    if (!this.max || max > this.max) {
-                        this.max = max;
-                        this.eqMax = false;
-                    }
-                } else if (dataType === XSD.custom("gYearMonth")) {
-                    const result = this.gYearMonthToMinMax(value);
-                    if (!result) return;
-                    const [min, max] = result;
-                    if (!this.min || min <= this.min) {
-                        this.min = min;
-                        this.eqMin = true;
-                    }
-                    if (!this.max || max > this.max) {
-                        this.max = max;
-                        this.eqMax = false;
-                    }
-                } else if (dataType === XSD.custom("date")) {
-                    const result = this.dateToMinMax(value);
-                    if (!result) return;
-                    const [min, max] = result;
-                    if (!this.min || min <= this.min) {
-                        this.min = min;
-                        this.eqMin = true;
-                    }
-                    if (!this.max || max > this.max) {
-                        this.max = max;
-                        this.eqMax = false;
-                    }
-                } else {
-                    // Check if it is a partial dateTime
-                    const result = this.partialDateTimeToMinMax(value);
-                    if (!result) return;
-                    const [min, max] = result;
-                    if (!this.min || min <= this.min) {
-                        this.min = min;
-                        this.eqMin = true;
-                    }
-                    if (!this.max || max > this.max) {
-                        this.max = max;
-                        this.eqMax = false;
-                    }
+            case TREE.custom("InBetweenRelation"): {
+                const between = parseInBetweenRelation(
+                    value,
+                    dataType,
+                    this.defaultTimezone,
+                );
+                if (between) {
+                    this.min = between.min;
+                    this.eqMin = true;
+                    this.max = between.max;
+                    this.eqMax = false;
```

Comment:
Here you don't do the detailed assignments. So this should go to the constructor, and the change from the constructor should go here.